pkg/daemon/daemon.go

```diff
@@ -484,35 +485,34 @@ func ReexecuteForTargetRoot(target string) error {
 	targetBinBase := "run/bin/machine-config-daemon"
 	targetBin := filepath.Join(target, targetBinBase)
 
-	// Be idempotent
-	if _, err := os.Stat(targetBin); err != nil {
-		sourceBinary := "/usr/bin/machine-config-daemon" + sourceBinarySuffix
-		src, err := os.Open(sourceBinary)
-		if err != nil {
-			return fmt.Errorf("opening %s: %w", sourceBinary, err)
-		}
-		defer src.Close()
+	targetBinDir := filepath.Dir(targetBin)
+	if err := os.MkdirAll(targetBinDir, 0o755); err != nil {
+		return fmt.Errorf("mkdir %s: %w", targetBinDir, err)
+	}
 
-		targetBinDir := filepath.Dir(targetBin)
-		if _, err := os.Stat(targetBinDir); err != nil {
-			if err := os.Mkdir(targetBinDir, 0o755); err != nil {
-				return fmt.Errorf("mkdir %s: %w", targetBinDir, err)
-			}
-		}
+	f, err := renameio.TempFile(targetBinDir, targetBin)
+	if err != nil {
+		return err
+	}
+	defer f.Cleanup()
 
-		f, err := os.Create(targetBin)
-		if err != nil {
-			return fmt.Errorf("writing %s: %w", targetBin, err)
-		}
-		if _, err := io.Copy(f, src); err != nil {
-			f.Close()
-			return fmt.Errorf("writing %s: %w", targetBin, err)
-		}
-		if err := f.Chmod(0o755); err != nil {
-			return err
-		}
-		// Must close our writable fd
+	sourceBinary := "/usr/bin/machine-config-daemon" + sourceBinarySuffix
```

Comment:
Functionally this should be the same as 4.14, but the ordering of code (and some other minor points) seem to be different from https://github.com/openshift/machine-config-operator/pull/3832

Did you want to consolidate the two in case we need to patch further?

Reply:
https://github.com/openshift/machine-config-operator/pull/3835